server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/TestCICSConnectProcessStatement.java

```diff
@@ -0,0 +1,88 @@
+        import org.eclipse.lsp4j.Position;
+        import org.eclipse.lsp4j.Range;
+        import org.junit.jupiter.api.Test;
+
+        import java.util.Map;
+
+/**
+ * Test CICS CONNECT PROCESS command. Documentation link: <a
+ * href="https://www.ibm.com/docs/en/cics-ts/6.x?topic=summary-connect-process"> Connect Process Command</a>
+ *
+ * <p>This class tests all variations of the Connect Process command found in the link above.
+ */
+public class TestCICSConnectProcessStatement {
+    private static final String CONNECT_PROCESS_VALID_PATH_ONE =
+            "CONNECT PROCESS CONVID({$varFour}) PROCNAME({$varFour}) PROCLENGTH({$varFour}) PIPLIST({$varFour}) "
+                    + "PIPLENGTH({$varFour}) SYNCLEVEL({$varFour}) STATE({$varFour})";
+    private static final String CONNECT_PROCESS_VALID_PATH_TWO =
+            "CONNECT PROCESS CONVID({$varFour}) PARTNER({$varFour}) "
+                    + "SYNCLEVEL({$varFour}) ";
+    private static final String CONNECT_PROCESS_INVALID =
+            "CONNECT PROCESS SESSION({$varFour}) SYNCLEVEL({$varFour}) {PARTNER | errorMissingProc | errorPartner}(100)";
+    private static final String CONNECT_PROCESS_PIPLENGTH_INVALID =
+            "CONNECT PROCESS CONVID({$varFour}) SYNCLEVEL({$varFour}) PROCNAME({$varFour})"
+                    + " {PIPLENGTH | errorPiplength }(100)";
+
+    @Test
+    void testAbendValid() {
+        CICSTestUtils.noErrorTest(CONNECT_PROCESS_VALID_PATH_ONE);
+    }
+    @Test
+    void testAbendValidAbcode() {
+        CICSTestUtils.noErrorTest(CONNECT_PROCESS_VALID_PATH_TWO);
+    }
+    @Test
+    void testConnectProcessSessionPartnerInvalid() {
+        Map<String, Diagnostic> expectedDiagnostic =
+                ImmutableMap.of(
+                        "errorMissingProc",
+                        new Diagnostic(
+                                new Range(new Position(13, 12), new Position(16, 24)),
+                                "Missing required option: PROCNAME",
+                                DiagnosticSeverity.Error,
+                                ErrorSource.PARSING.getText()),
+                        "errorPartner",
+                        new Diagnostic(
+                                new Range(new Position(16, 12), new Position(16,  19)),
+                                "Invalid option provided: PARTNER",
+                                DiagnosticSeverity.Error,
+                                ErrorSource.PARSING.getText()
+                                )
+                        );
+        CICSTestUtils.errorTest(CONNECT_PROCESS_INVALID, expectedDiagnostic);
+    }
+    @Test
+    void testConnectProcessPiplengthInvalid() {
+        Map<String, Diagnostic> expectedDiagnostic =
+                ImmutableMap.of(
+                        "errorPiplength",
+                        new Diagnostic(
+                                new Range(),
+                                "Extraneous input PIPLENGTH",
```

Comment:
I think a better message, indicating that PIPLIST is missing should be displayed.

Reply:
👍 Adressed in latest commit